cmd/cloud-controller-manager/app/apis/config/types.go

```diff
@@ -29,10 +29,10 @@ type CloudControllerManagerConfiguration struct {
 	metav1.TypeMeta
 
 	// Generic holds configuration for a generic controller-manager
-	Generic kubectrlmgrconfig.GenericControllerManagerConfiguration
+	Generic cmconfig.GenericControllerManagerConfiguration
 	// KubeCloudSharedConfiguration holds configuration for shared related features
 	// both in cloud controller manager and kube-controller manager.
-	KubeCloudShared kubectrlmgrconfig.KubeCloudSharedConfiguration
+	KubeCloudShared cmconfig.KubeCloudSharedConfiguration
```

Comment:
GenericControllerManagerConfiguration in k8s.io/controller-manager makes sense

KubeCloudSharedConfiguration and ServiceController types should go to k8s.io/cloud-provider, to avoid putting cloud-specific and kubernetes-specific stuff into the generic controller manager repo. KCM can depend on k8s.io/cloud-provider until it stops running cloud loops.

Reply:
Updated the PR to move KubeCloudSharedConfiguration and ServiceController types to k8s.io/cloud-provider. Please have a look when you got time😊